### PR TITLE
[feat/docs]: improve the `quick_sort.cpp` algorithm

### PR DESCRIPTION
**Pull request by Panquesito7**
#### Description of Change

- Improve the `quick_sort.cpp` algorithm.

#### Checklist

- [x] Added description of change
- [ ] ~Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)~
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: sorting/quick_sort.cpp [hunk 1]
```diff
@@ -19,83 +20,219 @@
  *      than x) before x, and put all greater elements (greater than x) after
  *      x. All this should be done in linear time
  *
+ * @author [David Leal](https://github.com/Panquesito7)
+ * @author Unknown author
```
Comment (tjgurwara99) [inline sorting/quick_sort.cpp:24]: Non-blocking; I think you can figure it out if you check out the commit `f22baf19210b17c8567d461ad35259322543701c` in your local - I don't have this repo cloned in my local env so can't do this. This was the parent commit before `ayankhan` made his changes to fix overflow issues, once you check that out you can do `git blame sorting/quick_sort.cpp` to check who wrote it 😄.

**Review by tjgurwara99 — DISMISSED** [review]

File: sorting/quick_sort.cpp (hunk 1 above)
Comment (Panquesito7) [reply sorting/quick_sort.cpp]: Nice, thanks! 👍 
I found the author's name within GitHub's UI. I just checked that commit, clicked on `Browse files`, searched for the quick sort file, and found it. However, the folders and the filename were renamed, thus, the history being rewritten/lost; thankfully, GitHub has the option to check older history, which then led me to the original author submitting the algorithm.

File: sorting/quick_sort.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@
  *      x. All this should be done in linear time
  *
  * @author [David Leal](https://github.com/Panquesito7)
- * @author Unknown author
+ * @author [popoapp](https://github.com/popoapp)
```
Comment (tjgurwara99) [inline sorting/quick_sort.cpp:24]: I don't think that's the author - this person joined GitHub very [recently](https://github.com/popoapp?tab=overview&from=2022-05-01&to=2022-05-31)...

Comment (Panquesito7) [reply sorting/quick_sort.cpp:24]: https://github.com/TheAlgorithms/C-Plus-Plus/commits/1c1746bc6496d568d572143cc5565dd3f6f884ee/Quick%20Sort.cpp?browsing_rename_history=true&new_path=sorting/Quick%20Sort.cpp&original_branch=f8da1b06852e33cb999b3df3fda8326b01500e90

Comment (tjgurwara99) [reply sorting/quick_sort.cpp:24]: Very weird indeed 😅

Comment (tjgurwara99) [reply sorting/quick_sort.cpp:24]: Though this seems legitimate - this person may have created a commit in the "git" repository (even though they didn't have an account on "GitHub") but this commit was pushed to "GitHub" by some other person who did have an account with write access. I guess this is related to the early days of `TheAlgorithms` organisation. I suspect that it wasn't an organisation and was turned into one after Anup Kumar Panwar already had a repository. So I presume that the author name is correct 😄

**Review by poyea — APPROVED** [review]

**Pull request merged**